lib/vsc/install/shared_setup.py

```diff
@@ -361,13 +373,15 @@ def get_name_url(self, filename=None, version=None, license_name=None):
         if len(res) != 3:
             raise Exception("Cannot determine name, url and download url from filename %s: got %s" % (filename, res))
         else:
+            keepers = {}
```

Comment:
The previous strategy was starting with a dictionary, res, and removing items from it **as it was being iterated through.** This triggers errors with testing, and generally isn't a good idea! To fix, I decided to add `keepers`, and instead of removing from res, I am adding entries from res to keepers that are intended to return. We then return "keepers" instead of res.

Reply:
To clarify: this is required because in Python 3, `.items()` returns an iterator, so modifying the dict we're iterating over means trouble in that case. In Python 2, `.items()` return a list (a copy of the contents of the dict), so modifying the original dict while iterating over `.items()` is fine in that case.

Reply:
thx!